lib/std/compress/deflate.zig

```diff
@@ -1,738 +1,30 @@
-                                // consistent with the use of readBitsNoEof.
-                                return error.EndOfStream;
-                            }
-                            self.window.appendUnsafe(tmp[0]);
-                        }
-
-                        if (N != length.*) {
-                            length.* -= N;
-                            return;
-                        }
-
-                        self.state = .DecodeBlockHeader;
-                    },
-                    .CopyLit => |c| {
-                        if (self.window.append(c) == 0) {
-                            return;
-                        }
-
-                        self.state = .DecodeBlockData;
-                    },
-                    .CopyFrom => |*info| {
-                        const written = self.window.copyFrom(info.distance, info.length);
-                        if (written != info.length) {
-                            info.length -= @truncate(u16, written);
-                            return;
-                        }
-
-                        self.state = .DecodeBlockData;
-                    },
-                }
-            }
-        }
-
-        fn init(source: ReaderType, window_slice: []u8) Self {
-            assert(math.isPowerOfTwo(window_slice.len));
-
-            return Self{
-                .inner_reader = source,
-                .window = .{ .buf = window_slice },
-                .seen_eos = false,
-                .state = .DecodeBlockHeader,
-                .hdist = undefined,
-                .hlen = undefined,
-                .bits = 0,
-                .bits_left = 0,
-            };
-        }
-
-        // Implements the io.Reader interface
-        pub fn read(self: *Self, buffer: []u8) Error!usize {
-            if (buffer.len == 0)
-                return 0;
-
-            // Try reading as much as possible from the window
-            var read_amt: usize = self.window.read(buffer);
-            while (read_amt < buffer.len) {
-                // Run the state machine, we can detect the "effective" end of
-                // stream condition by checking if any progress was made.
-                // Why "effective"? Because even though `seen_eos` is true we
-                // may still have to finish processing other decoding steps.
-                try self.step();
-                // No progress was made
-                if (self.window.readable() == 0)
-                    break;
-
-                read_amt += self.window.read(buffer[read_amt..]);
-            }
-
-            return read_amt;
-        }
-
-        pub fn reader(self: *Self) Reader {
-            return .{ .context = self };
-        }
-    };
-}
-
-pub fn inflateStream(reader: anytype, window_slice: []u8) InflateStream(@TypeOf(reader)) {
-    return InflateStream(@TypeOf(reader)).init(reader, window_slice);
-}
-
-test "lengths overflow" {
```

Comment:
Can the tests here please be maintained and adapted to the new API? Let's continue to pass these tests.

Reply:
Tests added in `compress/deflate/decompressor.zig`.